site/content/blog/2021-04-23-optimizing-go-compiler-1.md

```diff
@@ -0,0 +1,227 @@
+---
+title: "Fixing a bug in the Go compiler as a newbie: a deep dive (I)"
+heading: "Fixing a bug in the Go compiler as a newbie: a deep dive (I)"
+description: ""
+summary: "Contributing to the Go compiler is within everyone's reach. The first part of this series of posts defines the issue we are going to investigate, and introduces the parts of the Go compiler involved in its fix. We will learn about Static Single Assignment, rewrite rules and some specific incantations to hack on the Go compiler."
+slug: optimizing-go-compiler-1
+date: 2021-04-23T12:00:00-05:00
+categories:
+    - "go"
+author: Alejandro García Montoro
+github: agarciamontoro
+community: alejandro.garcia
+---
+
+[bf48163e8f2b604f3b9e83951e331cd11edd8495](https://github.com/golang/go/commit/bf48163e8f2b604f3b9e83951e331cd11edd8495). That is the hash of one of the 47791 commits that, at the time of this writing, build the whole git history of the Go compiler. It is definitely not the most important commit there, and most of the other ones probably deserve way more attention. But I like `bf48163`. It is a good commit, I think. I may be a bit biased, of course, as it is my first contribution to Go.
```

Comment:
> That is the hash of one of the 47791 commits that, at the time of this writing, build the whole git history of the Go compiler.

The last part reads a bit odd to me. I understand what you are trying to say, but what does "build the whole git history" mean? Isn't it just building the Go compiler?

0/5, but perhaps something like this is equivalent:

> [bf48163e8f2b604f3b9e83951e331cd11edd8495](https://github.com/golang/go/commit/bf48163e8f2b604f3b9e83951e331cd11edd8495). That is the hash of one of the 47791 commits of the Go programming language, at the time of this writing.

Reply:
> The last part reads a bit odd to me. I understand what you are trying to say, but what does "build the whole git history" mean? Isn't it just building the Go compiler?

My intention with this was to say that there are 47791 commits in the git history, each and everyone contributing to create the Go compiler. It is not build as in `make build`, but in the sense of constitute, of contributing to the history of the development of the compiler.

Would it be more clear to say something like this?

> That is the hash of one of the 47791 commits that, at the time of this writing, **form/populate/build up?** the whole git history of the Go compiler.

If it's still confusing, I'll use something like what you suggested.

Reply:
Right, that's more or less what I thought. It's the start of the post, so I'll leave it to whatever you feel is the best. :)

Reply:
Resolving this conversation too to remove some noise from the PR. Thanks again!